src/main/java/ch/uzh/ifi/hase/soprafs22/service/ChatService.java

```diff
@@ -1,34 +1,84 @@
 package ch.uzh.ifi.hase.soprafs22.service;
 
+import ch.uzh.ifi.hase.soprafs22.entity.*;
 import ch.uzh.ifi.hase.soprafs22.repository.ChatRepository;
-import ch.uzh.ifi.hase.soprafs22.repository.MatchRepository;
 import ch.uzh.ifi.hase.soprafs22.repository.MessageRepository;
-import ch.uzh.ifi.hase.soprafs22.repository.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
 import javax.transaction.Transactional;
-
+import java.util.ArrayList;
+import java.util.List;
 
 @Service
 @Transactional
 public class ChatService {
 
-  private final MatchRepository matchRepository;
-  private final UserRepository userRepository;
-  private final ChatRepository chatRepository;
-  private final MessageRepository messageRepository;
-
-  @Autowired
-  public ChatService(@Qualifier("MatchRepository")MatchRepository matchRepository,
-                     @Qualifier("userRepository") UserRepository userRepository,
-                     @Qualifier("ChatRepository") ChatRepository chatRepository,
-                     @Qualifier("MessageRepository") MessageRepository messageRepository) {
-
-    this.matchRepository = matchRepository;
-    this.userRepository = userRepository;
-    this.chatRepository = chatRepository;
-    this.messageRepository = messageRepository;
-  }
+    private final ChatRepository chatRepository;
+    private final MessageRepository messageRepository;
+
+    @Autowired
+    public ChatService( @Qualifier("ChatRepository") ChatRepository chatRepository,
+                        @Qualifier("MessageRepository") MessageRepository messageRepository) {
+
+        this.chatRepository = chatRepository;
+        this.messageRepository = messageRepository;
+    }
+
+    /**
+     * Get the first messages of all chats, if chat is empty, add null
+     * @param matches
+     * @return
+     */
+    public List<Message> getFirstMessages(List<Match> matches){
+
+        List<Message> firstMessages = new ArrayList<>();
+
+        for(Match match : matches){
+            // if no message, add null
+            try{
+            firstMessages.add(match.getChat().getMessages(0));
+            }catch(ArrayIndexOutOfBoundsException e) {
+                firstMessages.add(null);
```

Comment:
If you would do it like above, you wouldn't have to handle this exception (the client would get an empty list of messages, or a list with one element, which can be handled easily. (or you could do something like: 
`firstMessages.add((match.getChat().getMessages(0,0).isEmpty ? null : match.getChat().getMessages(0,0)`
(is this a valid expression in Java?) (Recommendation, Thinking loud)